### PR TITLE
Add dockerfile for edgesite

### LINKED ISSUE #875: Add Dockerfile for Edgesite
**What would you like to be added**:
Currently we do not have a dockerfile present for edgesite. There needs to be a Dockerfile added for edgesite which would be helpful to users who want to run edgesite as a container as opposed to running it as a binary.

**Why is this needed**:
A Dockerfile for edgesite would be helpful to users who want to run edgesite as a container as opposed to running it as a binary.

### PR DESCRIPTION
**Pull request by sujithsimon22**
**What type of PR is this?**
 /kind feature

**What this PR does / why we need it**:

This PR adds a Dockerfile for edgesite and a provision in makefile using which the user will be able to build a docker image for edgesite easily

**Which issue(s) this PR fixes**:

Fixes #875 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by sujithsimon22** [comment]
cc @kevin-wangzefeng  @rohitsardesai83 @sids-b

File: Makefile [hunk 1]
```diff
@@ -91,6 +91,17 @@ edgeimage:
 	--build-arg RUN_FROM=${ARCH}/docker:dind \
 	-f build/edge/Dockerfile .
 
+.PHONY: edgesiteimage
+edgesiteimage:
+	mkdir -p ./build/edgesite/tmp
+	rm -rf ./build/edgesite/tmp/*
```
Comment (rohitsardesai83) [inline Makefile]: first we create the tmp and then delete nothing inside it ?

Comment (sujithsimon22) [reply Makefile]: This is to handle cases in which the image has already been built before i.e. the tmp folder has contents inside. Something similar to this has been done while building edge_core image as well  https://github.com/kubeedge/kubeedge/blob/4d44da1ce4d9c001a814beaa44d0b32e257554db/Makefile#L86

Comment (rohitsardesai83) [reply Makefile]: then why are creating the directory ?

Comment (sujithsimon22) [reply Makefile]: the tmp folder does not exist by default, we create it the first we build the image and used for 
 downloading and using QEMU

**Pull request merged**